src/Map.ts

```diff
@@ -1,15 +1,53 @@
+import has from './support/has';
 import { ArrayLike } from './interfaces';
-import { hasClass } from './support/decorators';
 import global from './global';
 import { forOf, Iterable, IterableIterator, ShimIterator } from './iterator';
 import { is as objectIs } from './object';
 import './Symbol';
 
-export namespace Shim {
+export interface MapConstructor {
+	new (): Map<any, any>;
+	new <K, V>(entries?: [K, V][]): Map<K, V>;
+	new <K, V>(iterable: Iterable<[K, V]>): Map<K, V>;
+	readonly prototype: Map<any, any>;
+
+	[Symbol.species]: MapConstructor;
+}
+
+export default interface Map<K, V> {
+	/** Returns an iterable of entries in the map. */
+	[Symbol.iterator](): IterableIterator<[K, V]>;
+	[Symbol.toStringTag]: 'Map';
+
 	/**
-	 * An implementation analogous to the Map specification in ES2015.
+	 * Returns an iterable of key, value pairs for every entry in the map.
 	 */
-	export class Map<K, V> {
+	entries(): IterableIterator<[K, V]>;
+
+	clear(): void;
+	delete(key: K): boolean;
+	forEach(callbackfn: (value: V, key: K, map: Map<K, V>) => void, thisArg?: any): void;
+	get(key: K): V | undefined;
+
+	/**
+	 * Returns an iterable of keys in the map
+	 */
+	keys(): IterableIterator<K>;
+
+	has(key: K): boolean;
+	set(key: K, value: V): this;
+	readonly size: number;
+
+	/**
+	 * Returns an iterable of values in the map
+	 */
+	values(): IterableIterator<V>;
+}
+
+let ShimMap: MapConstructor;
```

Comment:
Can you `let ShimMap: MapConstructor = global.Map;`

Then replace if no `es6-map` and then just `export default ShimMap;`?